src/ir/utils.rs

```diff
@@ -408,9 +413,9 @@ where
     C: Ctx<T>,
 {
     /// A reference to the underlying value.
-    key: Idx<T>,
+    pub key: Idx<T>,
     /// A reference to the owner of the foreign key.
-    owner: Idx<C>,
+    pub owner: Idx<C>,
```

Comment:
Don't make these `pub`; that allows someone to mutate them. Instead, let's provide accessor methods.